frappe/desk/reportview.py

```diff
@@ -18,20 +19,157 @@
+
+	data.strict = None
 
-	is_report = data.get('view') == 'Report'
+	return data
+
+def validate_fields(data):
+	wildcard = update_wildcard_field_param(data)
+
+	for field in data.fields or []:
+		fieldname = extract_fieldname(field)
+		if is_standard(fieldname):
+			continue
+
+		meta, df = get_meta_and_docfield(fieldname, data)
 
+		if not df:
+			if wildcard:
+				continue
+			else:
+				raise_invalid_field(fieldname)
+
+		# remove the field from the query if the report hide flag is set and current view is Report
+		if df.report_hide and data.view == 'Report':
+			data.fields.remove(field)
+			continue
+
+		if df.fieldname in [_df.fieldname for _df in meta.get_high_permlevel_fields()]:
+			if df.get('permlevel') not in meta.get_permlevel_access(parenttype=data.doctype):
+				data.fields.remove(field)
+
+def validate_filters(data, filters):
+	if isinstance(filters, list):
+		# filters as list
+		for condition in filters:
+			if len(condition)==3:
+				# [fieldname, condition, value]
+				fieldname = condition[0]
+				if is_standard(fieldname):
+					continue
+				meta, df = get_meta_and_docfield(fieldname, data)
+				if not df:
+					raise_invalid_field(condition[0])
+			else:
+				# [doctype, fieldname, condition, value]
+				fieldname = condition[1]
+				if is_standard(fieldname):
+					continue
+				meta = frappe.get_meta(condition[0])
+				if not meta.get_field(fieldname):
+					raise_invalid_field(fieldname)
+
+	else:
+		for fieldname in filters:
+			if is_standard(fieldname):
+				continue
+			meta, df = get_meta_and_docfield(fieldname, data)
+			if not df:
+				raise_invalid_field(fieldname)
+
+def setup_group_by(data):
+	'''Add columns for aggregated values e.g. count(name)'''
+	if data.group_by:
+		if data.aggregate_function.lower() not in ('count', 'sum', 'avg'):
+			frappe.throw('Invalid aggregate function')
+		if '`' in data.aggregate_on:
+			raise_invalid_field(data.aggregate_on)
+		data.fields.append('{aggregate_function}(`tab{doctype}`.`{aggregate_on}`) AS _aggregate_column'.format(**data))
+		if data.aggregate_on:
+			data.fields.append(data.aggregate_on)
+
+		data.pop('aggregate_on')
+		data.pop('aggregate_function')
+
+def raise_invalid_field(fieldname):
+	frappe.throw(_('Field not permitted in query') + ': {0}'.format(fieldname), frappe.DataError)
+
+def is_standard(fieldname):
+	if '.' in fieldname:
+		parenttype, fieldname = get_parenttype_and_fieldname(fieldname, None)
+	return fieldname in default_fields or fieldname in optional_fields
+
+def extract_fieldname(field):
+	for text in (',', '/*', '#'):
+		if text in field:
+			raise_invalid_field(field)
+
+	fieldname = field
+	for sep in (' as ', ' AS '):
+		if sep in fieldname:
+			fieldname = fieldname.split(sep)[0]
+
+	# certain functions allowed, extract the fieldname from the function
+	if (fieldname.startswith('count(')
+		or fieldname.startswith('sum(')
+		or fieldname.startswith('avg(')):
+		if not fieldname.strip().endswith(')'):
```

Comment:
May be a list of agregator_functions? 
allowed_aggregators = ['count', 'sum', 'avg']